lib/install/management/upgrade.go

```diff
@@ -233,7 +224,17 @@ func (d *Dispatcher) ensureRollbackReady(conf *config.VirtualContainerHostConfig
 		log.Infof("Roll back finished - Appliance is kept in powered off status")
 		return nil
 	}
-	return d.startAppliance(conf)
+	if err = d.startAppliance(conf); err != nil {
+		return err
+	}
+
+	ctx, cancel := context.WithTimeout(d.ctx, settings.Timeout)
+	defer cancel()
+	if err = d.CheckServiceReady(ctx, conf, nil); err != nil {
+		// do not return error in this case, to make sure clean up continues
+		log.Info("\tAPI may be slow to start - try to connect to API after a few minutes")
```

Comment:
You don't need to check if err == deadline exceeded here?  you do so in other places.